starter-code/TicTacToe/app/src/main/java/ly/generalassemb/drewmahrt/tictactoe/MainActivity.java

```diff
@@ -1,17 +1,154 @@
 package ly.generalassemb.drewmahrt.tictactoe;
 
+import android.content.DialogInterface;
 import android.content.Intent;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.support.v7.app.AlertDialog;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
 import android.view.View;
 import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
 import android.widget.Toast;
 
+import java.util.ArrayList;
+
 public class MainActivity extends AppCompatActivity {
+    EditText p1EditText, p2EditText;
+    Button startGameButton, clearHistButton;
+    TextView lastWinText;
+    public static final int OUTCOME_CODE = 2369;
+    public static final int WINNER_CODE= 1337;
+    public static final int DRAW_CODE = 424;
+    public static SQLiteDatabase gameHistoryDB;
+    public Cursor c;
+    ArrayList<GameResult> gameResults;
+    RecyclerView resultsRecycler;
+
+    //ToDo: Add an activity to display gamestate results of previous games.
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        gameResults = new ArrayList<>();
+
+        gameHistoryDB = openOrCreateDatabase("Game_History", MODE_PRIVATE, null);
+        gameHistoryDB.execSQL("CREATE TABLE IF NOT EXISTS game_history (gameID INTEGER PRIMARY KEY, p1Name VARCHAR, p2Name VARCHAR, winner VARCHAR, gameState VARCHAR)");
+        //If game was draw winner var is "draw"
+        c = gameHistoryDB.rawQuery("SELECT * FROM game_history", null);
+        c.moveToFirst();
```

Comment:
Kudos for making the database work! Tomorrow we'll cover SQLiteOpenHelper, which will make working with a database much simpler.